runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManagerOptions.java

```diff
@@ -0,0 +1,37 @@
+package org.jrd.backend.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArchiveManagerOptions {
+
+	private static class ArchiveManagerOptionsHolder {
+		private static final ArchiveManagerOptions INSTANCE = new ArchiveManagerOptions();
+	}
+
+	public static ArchiveManagerOptions getInstance() {
+		return ArchiveManagerOptionsHolder.INSTANCE;
+	}
+
+	private final List<String> extensions = new ArrayList<>();
+	private static final List<String> defaults = List.of(".zip", ".jar", ".war", ".ear");
+
+	public void addExtension(String s) {
+		if (s.startsWith(".")) {
+			extensions.add(s);
+		} else {
+			extensions.add("." + s);
+		}
+	}
+
+	public void clearExtensions() {
+		extensions.clear();
+	}
+
+	public List<String> getExtensions() {
+		if (extensions.isEmpty()) {
+			return defaults;
+		}
+		return extensions;
```

Comment:
all look good now, except get/set clear.

Pls remove clear.
Pls change getter and setter to "normal and obvious ones"
pls create method of " ArchiveManagerOptions.getInstance().isSupportedArchive(file)" Which will do the split and dots work

Great work! Thanx a lot!